### PR TITLE
Revert #4725 - MIDI use logical time on macOS

### LINKED ISSUE #5622: MIDI issues on macOS
## Environment

* SuperCollider version: 3.12.1
* Operating system: macOS 10.15 (Catalina), macOS 11.6 (Big Sur)
* Other details (Qt version, audio driver, etc.):

## Steps to reproduce

```supercollider
MIDIClient.init;
m = MIDIOut.newByName("X-TOUCH COMPACT", "X-TOUCH COMPACT"); // testing with a midi cotroller
m.latency; // reports the default 0.2

m.noteOn(0, 30, 127); // this turns button LED on
// after the first start of SC after the boot, there's about 2s delay for the message to be sent (Catalina and Big Sur)
m.noteOn(0, 30, 0); // turn it off

// put computer to sleep for about a minute
m.noteOn(0, 30, 127); // now the message is delayed by over 40 seconds (the time the laptop was physically asleep) - Big Sur only
```

## TL;DR
As much as I'd like to keep the improved MIDI timing introduced in #4725, that PR seems to introduce issues that IMO are hard to be solved. Unless someone has an idea how to tackle them, I'd suggest to revert #4725 (and possibly release 3.12.2, if desired).

## Expected vs. actual behavior

#4725 improved the relative timing of MIDI messages, but introduced 2 (3) new issues:
1. all MIDI messages are delayed approx. 2s after the first run of `MIDIClient.init` after SC starts; re-running `MIDIClient.init` removes that delay. This happens on Catalina and on Big Sur
2. MIDI messages are additionally delayed by the time the computer was asleep since running `MIDIClient.init`. This happens on Big Sur only
3. If we use logical time for MIDI messages, latency calculation should probably be revised (right now the latency value is just added to the resulting logical timestamp). Depending how we deal with the above issues, this point might be moot if we revert #4725.

Details:
#4725 introduces definition of "starting time" using both SC thread time and time from `AudioHostGetTime()`. This seems like a good approach in principle. The issue is that starting with Big Sur, the time from `AudioHostGetTime()` does not advance when the computer is asleep, while the thread clock does advance. This conceptually similar to the issue solved in #5298 (also see explanation [here](https://github.com/supercollider/supercollider/issues/5168#issuecomment-749277265)) since that issue also dealt with comparing times from different references. Since MIDI message on macOS expects the time as reported from `AudioHostGetTime()`, I don't know how we could reliably calculate logical time, since the SC thread time does advance while the computer is asleep. I guess we could conceivably re-set that offset every time system wakes from sleep, but I'm not sure how to go about it or whether that's the best solution.

Regarding problem `1.`, there seems to be some fluctuation in thread timing that occurs only on the first MIDI initialization after SC starts. I don't know where that comes from.
I tried posting the offset between `AudioHostGetTime()` and SC thread when initializing MIDI and that offset is consistently off by approx. 2s between the first and the subsequent runs of `MIDIClient.init`. The subsequent runs report offsets that differ only by few ms, which is to be expected, I think, and is not an issue. Only that initial offset is considerably large. (See below for examining that delay)

I was trying to fix the issue with clock diverging when computer goes to sleep, but that didn't work since we ultimately need to use the time in relation to `AudioHostGetTime()` to schedule MIDI events. However, my code does post initial time difference between SC clock and system nanosecond clock, where one can observe the difference in offsets between the first and the subsequent initializations of MIDI client: https://github.com/dyfer/supercollider/tree/topic/macos-midi-01

### PR DESCRIPTION
**Pull request by dyfer**
This reverts commit c6790a9168be1972724034e79f83d53ef4f0399c, reversing changes made to 13be6e8202674a98d0e20b58201ec10763515de3.

## Purpose and Motivation

Fixes #5622

See linked issue for explanation and details.
I also cherry-picked CI commit that fixes the macOS legacy build.

I'm suggesting to merge this into `3.12` branch in order to release `3.12.2` with this fix, but I'm open to suggestions about this.

## Types of changes

- Bug fix

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -326,6 +326,8 @@ jobs:
       - name: install qt using aqtinstall
         uses: jurplel/install-qt-action@v2
         if: matrix.qt-version
+        env:
```
Comment (jamshark70) [inline .github/workflows/actions.yml:329]: This appears to be a new change, not a reversion.

**Review by jamshark70 — CHANGES_REQUESTED** [review]
Just noting an unrelated change that leaked in. As for the CoreMIDI changes, I can't comment.

File: .github/workflows/actions.yml (hunk 1 above)
Comment (dyfer) [reply .github/workflows/actions.yml:329]: As I noted above

> I also cherry-picked CI commit that fixes the macOS legacy build.

This is a fix from #5627 - if we want to merge this into 3.12 and make another release, we need that fix for the macOS legacy build.

Comment (jamshark70) [reply .github/workflows/actions.yml:329]: OK, I've withdrawn that review. Thanks (and sorry that I missed / didn't understand the initial comment).

**Review by joshpar — APPROVED** [review]
